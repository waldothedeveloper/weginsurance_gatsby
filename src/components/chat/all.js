import React from "react";

export const Chat = () => (
  <div>
    <h1>This is the CHAT component</h1>
  </div>
);
