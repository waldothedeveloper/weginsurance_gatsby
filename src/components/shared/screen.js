import React from "react";
import propTypes from "prop-types";
export const SharedScreen = ({
  rightPanel: RightPanel,
  leftPanel: LeftPanel,
}) => (
  <div className="z-0 mx-auto grid grid-cols-12 gap-8 px-6">
    <div className="col-span-4 h-full rounded-lg border-2 border-solid border-gray-100 py-6 lg:px-4">
      {LeftPanel}
    </div>
    <div className="col-span-8 h-full rounded-lg border-2 border-solid border-gray-100 pt-6 lg:px-1">
      {RightPanel}
    </div>
  </div>
);

SharedScreen.propTypes = {
  rightPanel: propTypes.node.isRequired,
  leftPanel: propTypes.node.isRequired,
};
