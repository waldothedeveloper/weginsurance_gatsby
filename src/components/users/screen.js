import { EditUserForm } from "./editUserForm";
import { EmptyUserList } from "./emptyUserList";
import { NewUserForm } from "./newUserForm";
import { NoUsers } from "./empty";
import React from "react";
import { UserDetails } from "./details";
import { UserScreenLoader } from "../shared/loaders/usersScreenLoader";
import { UsersDictionary } from "./usersDictionary";
import { useCreateUser } from "../../hooks/user/useCreateUser";
import { useEditUser } from "../../hooks/user/useEditUser";
import { useReadUsers } from "../../hooks/user/useReadUsers";
import { useSwitcherUserCRUDMode } from "../../hooks/user/useSwitcherUserCRUDMode";

export const UsersScreen = () => {
  const { users } = useReadUsers();

  const {
    selectedUser,
    editUser,
    newUser,
    handleEditUser,
    handleNewUser,
    handleSelectUser,
    handleUserCRUDAction,
    handleUpdatedUser,
  } = useSwitcherUserCRUDMode();

  const { values, handleChange, handleSubmit, errors } = useCreateUser();
  const {
    userToEdit,
    editUserErrors,
    handleEditUserChange,
    handleUserEditSubmit,
  } = useEditUser(selectedUser, handleUpdatedUser);

  return (
    <div className="z-0 mx-auto grid grid-cols-12 gap-8 px-6">
      <div className="col-span-4 h-screen overflow-y-auto rounded-lg border-2 border-solid border-gray-100 py-6 lg:px-4">
        {users && users.length > 0 ? (
          <UsersDictionary
            users={users}
            handleUser={handleNewUser}
            handleSelectUser={handleSelectUser}
          />
        ) : users && users.length === 0 ? (
          <EmptyUserList createUser={handleNewUser} />
        ) : (
          <UserScreenLoader />
        )}
      </div>
      <div className="col-span-8 h-screen overflow-y-auto rounded-lg border-2 border-solid border-gray-100 pb-40 lg:px-4">
        {newUser ? (
          <NewUserForm
            values={values}
            handleChange={handleChange}
            handleSubmit={handleSubmit}
            errors={errors}
            handleUser={handleUserCRUDAction}
          />
        ) : selectedUser ? (
          <UserDetails
            selectedUser={selectedUser}
            handleSelectedUser={handleSelectUser}
            handleEditUser={handleEditUser}
            users={users}
            handleUser={handleUserCRUDAction}
          />
        ) : editUser ? (
          <EditUserForm
            values={userToEdit}
            handleChange={handleEditUserChange}
            handleSubmit={handleUserEditSubmit}
            errors={editUserErrors}
            handleUser={handleUserCRUDAction}
          />
        ) : (
          <NoUsers createUser={handleNewUser} />
        )}
      </div>
    </div>
  );
};
