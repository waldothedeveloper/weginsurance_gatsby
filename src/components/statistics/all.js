import React from "react";
import { useGetAndDeleteConversations } from "../../hooks/chat/useGetAndDeleteConversations";
export const Statistics = () => (
  // const { data, error } = useGetAndDeleteConversations();
  // console.log(`error: `, error);
  // console.log(`ALL COVERSARIONS DELETED: `, data);

  <div>I am the Statistics Component</div>
);
