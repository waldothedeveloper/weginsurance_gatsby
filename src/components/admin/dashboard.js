import { BellIcon, MenuAlt2Icon, XIcon } from "@heroicons/react/outline";
import { Dialog, Transition } from "@headlessui/react";
import { Fragment, useState } from "react";

import { Profile } from "./profile";
import React from "react";
import { Screen } from "./screen";
import { Search } from "./search";
import { StaticImage } from "gatsby-plugin-image";
import { Switcher } from "./switcher";
import { useSwitcher } from "../../hooks/dashboard/useSwitcher";

const Dashboard = () => {
  const [sidebarOpen, setSidebarOpen] = useState(false);
  const { module, handleModuleSwitch } = useSwitcher();

  return (
    <div className="flex h-screen min-h-screen flex-col">
      <Transition.Root show={sidebarOpen} as={Fragment}>
        <Dialog
          as="div"
          className="fixed inset-0 z-40 flex md:hidden"
          onClose={setSidebarOpen}
        >
          <Transition.Child
            as={Fragment}
            enter="transition-opacity ease-linear duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="transition-opacity ease-linear duration-300"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <Dialog.Overlay className="fixed inset-0 bg-gray-600 bg-opacity-75" />
          </Transition.Child>
          <Transition.Child
            as={Fragment}
            enter="transition ease-in-out duration-300 transform"
            enterFrom="-translate-x-full"
            enterTo="translate-x-0"
            leave="transition ease-in-out duration-300 transform"
            leaveFrom="translate-x-0"
            leaveTo="-translate-x-full"
          >
            <div className="relative flex w-full max-w-xs flex-1 flex-col bg-sky-700 pt-5 pb-4">
              <Transition.Child
                as={Fragment}
                enter="ease-in-out duration-300"
                enterFrom="opacity-0"
                enterTo="opacity-100"
                leave="ease-in-out duration-300"
                leaveFrom="opacity-100"
                leaveTo="opacity-0"
              >
                <div className="absolute top-0 right-0 -mr-12 pt-2">
                  <button
                    type="button"
                    className="ml-1 flex h-10 w-10 items-center justify-center rounded-full focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                    onClick={() => setSidebarOpen(false)}
                  >
                    <span className="sr-only">Close sidebar</span>
                    <XIcon className="h-6 w-6 text-white" aria-hidden="true" />
                  </button>
                </div>
              </Transition.Child>
              <div className="flex flex-shrink-0 items-center justify-center px-4">
                <StaticImage
                  placeholder="blurred"
                  layout="fullWidth"
                  className="h-20 w-20 rounded-full"
                  src="../../images/weginsurance_logo.jpeg"
                  alt="girl smiling representing a bright future for the weginsurance's company"
                />
              </div>
              <div className="mt-5 h-0 flex-1 overflow-y-auto">
                <nav className="space-y-1 px-2">
                  <Switcher handleModule={handleModuleSwitch} module={module} />
                </nav>
              </div>
            </div>
          </Transition.Child>
          {/* Dummy element to force sidebar to shrink to fit close icon */}
          <div className="w-14 flex-shrink-0" aria-hidden="true" />
        </Dialog>
      </Transition.Root>

      {/* Static sidebar for desktop */}
      <div className="hidden md:fixed md:inset-y-0 md:flex md:w-64 md:flex-col">
        {/* Sidebar component, swap this element with another sidebar if you like */}
        <div className="flex flex-grow flex-col overflow-y-auto bg-sky-700 pt-5">
          <div className="flex flex-shrink-0 items-center justify-center px-4">
            <StaticImage
              placeholder="blurred"
              layout="fullWidth"
              className="h-20 w-20 rounded-full"
              src="../../images/weginsurance_logo.jpeg"
              alt="weginsurance logo"
            />
          </div>
          <div className="mt-5 flex flex-1 flex-col">
            <nav className="flex-1 space-y-1 px-2 pb-4">
              <Switcher handleModule={handleModuleSwitch} module={module} />
            </nav>
          </div>
        </div>
      </div>
      <div className="flex flex-1 flex-col md:pl-64">
        <div className="sticky top-0 z-40 flex h-16 flex-shrink-0 bg-white shadow">
          <button
            type="button"
            className="border-r border-gray-200 px-4 text-gray-500 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-sky-500 md:hidden"
            onClick={() => setSidebarOpen(true)}
          >
            <span className="sr-only">Open sidebar</span>
            <MenuAlt2Icon className="h-6 w-6" aria-hidden="true" />
          </button>
          <div className="flex flex-1 justify-between px-4">
            <Search />
            <div className="ml-4 flex items-center md:ml-6">
              <button
                type="button"
                className="rounded-full bg-white p-1 text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
              >
                <span className="sr-only">View notifications</span>
                <BellIcon className="h-6 w-6" aria-hidden="true" />
              </button>

              {/* Profile dropdown */}
              <Profile />
            </div>
          </div>
        </div>

        {/* render screen component here */}
        <Screen module={module} />
      </div>
    </div>
  );
};

export default Dashboard;
