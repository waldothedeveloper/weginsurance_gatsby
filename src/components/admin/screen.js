import React, { useEffect, useState } from "react";

import { Chat } from "../chat/index";
import PropTypes from "prop-types";
import { Statistics } from "../statistics/all";
import { UsersScreen } from "../users/index";

const Skeleton = ({ children }) => <main className="py-10">{children}</main>;

export const Screen = ({ module }) => {
  const [componentToRender, setComponentToRender] = useState(`Dashboard`);

  useEffect(() => {
    const currentModule = module.filter((item) => item.current)[0].name;
    setComponentToRender(currentModule);
  }, [module]);

  switch (componentToRender) {
    case `Dashboard`:
      return (
        <Skeleton>
          <Statistics />
        </Skeleton>
      );
    case `Clientes`:
      return (
        <Skeleton>
          <UsersScreen />
        </Skeleton>
      );
    case `Chat`:
      return (
        <Skeleton>
          <Chat />
        </Skeleton>
      );
    default:
      return (
        <Skeleton>
          <Statistics />
        </Skeleton>
      );
  }
};

Screen.propTypes = {
  module: PropTypes.array.isRequired,
};

Skeleton.propTypes = {
  children: PropTypes.node.isRequired,
};
