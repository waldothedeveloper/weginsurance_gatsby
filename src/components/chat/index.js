import { ChatWall } from "./wall";
import { Error } from "./error";
import { NoUserSelected } from "./noUserSelected";
import React from "react";
import { SharedScreen } from "../shared/screen";
import { Spinner } from "../shared/spinner";
import { Transition } from "@headlessui/react";
import { UserScreenLoader } from "../shared/loaders/usersScreenLoader";
import { UsersList } from "./usersList";
import { useConfigureChat } from "../../hooks/chat/useConfigureChat";
import { useReadUsers } from "../../hooks/user/useReadUsers";
import { useSelectUserToChat } from "../../hooks/chat/useSelectUserToChat";

//
export const Chat = () => {
  const { users } = useReadUsers();
  const { participantInfo, handleParticipantInfo } = useSelectUserToChat();

  const { statusString, messages, newMessage, handleChange, handleSubmit } =
    useConfigureChat(participantInfo);

  return (
    <SharedScreen
      leftPanel={
        <Transition
          show={true}
          enter="transition-opacity duration-75"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="transition-opacity duration-150"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          {users && users.length > 0 ? (
            <div className="flex flex-1 flex-col overflow-hidden">
              <UsersList
                users={users}
                chooseParticipant={handleParticipantInfo}
              />
            </div>
          ) : users && users.length === 0 ? (
            <div>WE NEED SOMETHING HERE</div>
          ) : (
            <UserScreenLoader chat={true} />
          )}
        </Transition>
      }
      rightPanel={
        statusString === `You are connected.` ? (
          <ChatWall
            messages={messages}
            handleChange={handleChange}
            handleSubmit={handleSubmit}
            newMessage={newMessage}
          />
        ) : statusString === `Connecting to Twilio…` ? (
          <Spinner>Conectando el chat...</Spinner>
        ) : statusString === `Failed to connect.` ? (
          <Error />
        ) : statusString === `Disconnecting from Twilio…` ? (
          <Spinner>Desconectando el chat...</Spinner>
        ) : (
          <NoUserSelected />
        )
      }
    />
  );
};
