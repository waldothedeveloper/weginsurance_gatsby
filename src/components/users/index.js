import { EditUserForm } from "./editUserForm";
import { EmptyUserList } from "./emptyUserList";
import { NewUserForm } from "./newUserForm";
import { NoUsers } from "./empty";
import React from "react";
import { SharedScreen } from "../shared/screen";
import { Transition } from "@headlessui/react";
import { UserDetails } from "./details";
import { UserScreenLoader } from "../shared/loaders/usersScreenLoader";
import { UsersDictionary } from "./usersDictionary";
import { useCreateUser } from "../../hooks/user/useCreateUser";
import { useEditUser } from "../../hooks/user/useEditUser";
import { useReadUsers } from "../../hooks/user/useReadUsers";
import { useSwitcherUserCRUDMode } from "../../hooks/user/useSwitcherUserCRUDMode";

export const UsersScreen = () => {
  const { users } = useReadUsers();

  const {
    selectedUser,
    editUser,
    newUser,
    handleEditUser,
    handleNewUser,
    handleSelectUser,
    handleUserCRUDAction,
    handleUpdatedUser,
  } = useSwitcherUserCRUDMode();

  const { values, handleChange, handleSubmit, errors } = useCreateUser();
  const {
    userToEdit,
    editUserErrors,
    handleEditUserChange,
    handleUserEditSubmit,
  } = useEditUser(selectedUser, handleUpdatedUser);

  return (
    <SharedScreen
      leftPanel={
        <Transition
          show={true}
          enter="transition-opacity duration-75"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="transition-opacity duration-150"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          {users && users.length > 0 ? (
            <UsersDictionary
              users={users}
              handleUser={handleNewUser}
              handleSelectUser={handleSelectUser}
            />
          ) : users && users.length === 0 ? (
            <EmptyUserList createUser={handleNewUser} />
          ) : (
            <UserScreenLoader />
          )}
        </Transition>
      }
      rightPanel={
        <Transition
          show={true}
          enter="transition-opacity duration-75"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="transition-opacity duration-150"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          {newUser ? (
            <div className="h-[82vh] flex-none overflow-y-scroll px-2">
              <NewUserForm
                values={values}
                handleChange={handleChange}
                handleSubmit={handleSubmit}
                errors={errors}
                handleUser={handleUserCRUDAction}
              />
            </div>
          ) : selectedUser ? (
            <UserDetails
              selectedUser={selectedUser}
              handleSelectedUser={handleSelectUser}
              handleEditUser={handleEditUser}
              users={users}
              handleUser={handleUserCRUDAction}
            />
          ) : editUser ? (
            <div className="h-[82vh] flex-none overflow-y-scroll px-2">
              <EditUserForm
                values={userToEdit}
                handleChange={handleEditUserChange}
                handleSubmit={handleUserEditSubmit}
                errors={editUserErrors}
                handleUser={handleUserCRUDAction}
              />
            </div>
          ) : (
            <NoUsers createUser={handleNewUser} />
          )}
        </Transition>
      }
    />
  );
};
