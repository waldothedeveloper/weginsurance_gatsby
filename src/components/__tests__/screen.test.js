import { FolderIcon, HomeIcon, UsersIcon } from "@heroicons/react/outline";
import { render, screen } from "@testing-library/react";

import { Chat } from "../../components/chat/all";
import React from "react";
import { Screen } from "../admin/screen";
import { Statistics } from "../../components/statistics/all";
import { UsersScreen } from "../../components/users/screen";

//
const module = [
  {
    name: `Dashboard`,
    icon: HomeIcon,
    current: true,
    comp: Statistics,
  },
  {
    name: `Clientes`,
    icon: UsersIcon,
    current: false,
    comp: UsersScreen,
  },
  { name: `Chat`, icon: FolderIcon, current: false, comp: Chat },
];

//
describe(`Screen`, () => {
  it(`should render correctly`, () => {
    const { container } = render(<Screen module={module} />);
    expect(container).toMatchSnapshot();
  });

  it(`should render the current module`, () => {
    render(<Screen module={module} />);
    expect(screen.getByText(/statistics/i)).toBeInTheDocument();
  });
});
