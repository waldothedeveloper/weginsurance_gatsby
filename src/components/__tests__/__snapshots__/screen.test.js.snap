// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen should render correctly 1`] = `
<div>
  <main
    class="py-10"
  >
    <div>
      This is the Statistics WHAT
    </div>
  </main>
</div>
`;
